File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples-js/dedicatedHostExamples/DedicatedHost_Restart.js
const { ComputeManagementClient } = require("@azure/arm-compute");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Restart the dedicated host. The operation will complete successfully once the dedicated host has restarted and is running. To determine the health of VMs deployed on the dedicated host after the restart check the Resource Health Center in the Azure Portal. Please refer to https://docs.microsoft.com/azure/service-health/resource-health-overview for more details.
 *
 * @summary Restart the dedicated host. The operation will complete successfully once the dedicated host has restarted and is running. To determine the health of VMs deployed on the dedicated host after the restart check the Resource Health Center in the Azure Portal. Please refer to https://docs.microsoft.com/azure/service-health/resource-health-overview for more details.
 * x-ms-original-file: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/dedicatedHostExamples/DedicatedHost_Restart.json
 */
async function restartDedicatedHost() {
  const subscriptionId = "{subscription-id}";
  const resourceGroupName = "myResourceGroup";
  const hostGroupName = "myDedicatedHostGroup";
  const hostName = "myHost";
  const credential = new DefaultAzureCredential();
  const client = new ComputeManagementClient(credential, subscriptionId);
  const result = await client.dedicatedHosts.beginRestartAndWait(
    resourceGroupName,
    hostGroupName,
    hostName
  );
  console.log(result);
}

restartDedicatedHost().catch(console.error);

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-03-01/examples-java/dedicatedHostExamples/DedicatedHost_Restart.java
import com.azure.core.util.Context;

/** Samples for DedicatedHosts Restart. */
public final class Main {
    /*
     * x-ms-original-file: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-03-01/examples/dedicatedHostExamples/DedicatedHost_Restart.json
     */
    /**
     * Sample code: Restart Dedicated Host.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void restartDedicatedHost(com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .virtualMachines()
            .manager()
            .serviceClient()
            .getDedicatedHosts()
            .restart("myResourceGroup", "myDedicatedHostGroup", "myHost", Context.NONE);
    }
}
